lib/Basic/SourceLoc.cpp

```diff
@@ -24,7 +24,11 @@ void SourceManager::verifyAllBuffers() const {
   };
 
   // FIXME: This depends on the buffer IDs chosen by llvm::SourceMgr.
-  __attribute__((used)) static char arbitraryTotal = 0;
+  // _attribute__((used) is not defined in MSVC
+#if !defined(_MSC_VER)
+  __attribute__((used))
+#endif
```

Comment:
This is the only place `__attribute((used))` is used in the entire library I think, so I'm not sure its worth the effort!